File: app/src/main/java/com/sofar/network/ApiService.java
package com.sofar.network;

import com.sofar.network.response.Response;

import io.reactivex.Observable;
import retrofit2.http.GET;

public interface ApiService {

  @GET("/banner/json")
  Observable<String> getBannerData();

  @GET("/banner/json")
  Observable<Response<String>> getBannerDataResponse();

}

File: framework/network/src/main/java/com/sofar/network/openapi/api/ApiService.kt
package com.sofar.network.openapi.api

import com.sofar.network.openapi.api.model.Banner
import retrofit2.http.GET

interface ApiService {

  @GET("/banner/json")
  suspend fun getBannerDataResponse(): Result<String>

  @GET("/banner/json")
  suspend fun getBannerData(): Result<ApiResponse<List<Banner>>>
}
